From 6234026ff24e2cb68ff582096d8e9ff1069dc294 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 2 Jan 2018 15:40:19 +0000
Subject: [PATCH 03/18] Win32: Do not link -static-libgcc

---
diff --git a/src/gnuwin32/fixed/etc/Makeconf b/src/gnuwin32/fixed/etc/Makeconf
index d54c158..b8ac01a 100644
--- a/src/gnuwin32/fixed/etc/Makeconf
+++ b/src/gnuwin32/fixed/etc/Makeconf
@@ -11,8 +11,8 @@ else
   DEBUGFLAG=
 endif
 
-DLLFLAGS += -static-libgcc
-LINKFLAGS += -static-libgcc
+DLLFLAGS+= 
+LINKFLAGS+= 
 
 ## The rtools40 installer sets RTOOLS40_HOME, default to standard install path
 RTOOLS40_HOME ?= c:/rtools40
-- 
2.20.1

