From 461f2f1ef08c05d06eeb95908f86799f517fa76f Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 2 Jan 2018 15:40:19 +0000
Subject: [PATCH 03/13] Win32: Do not link -static-libgcc

---
 src/gnuwin32/fixed/etc/Makeconf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gnuwin32/fixed/etc/Makeconf b/src/gnuwin32/fixed/etc/Makeconf
index 41df5e0..3838f12 100644
--- a/src/gnuwin32/fixed/etc/Makeconf
+++ b/src/gnuwin32/fixed/etc/Makeconf
@@ -16,8 +16,8 @@ else
   FDEBUGFLAG=
 endif
 
-DLLFLAGS += -static-libgcc
-LINKFLAGS += -static-libgcc
+DLLFLAGS +=
+LINKFLAGS +=
 
 
 ## Things which are substituted by fixed/Makefile (and also -O3 -> -O2)
-- 
2.39.3 (Apple Git-146)

