From 4b9f9c7eadb839835e3a50315c0f52e935c8222b Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 2 Jan 2018 15:40:19 +0000
Subject: [PATCH 03/16] Win32: Do not link -static-libgcc

---
 src/gnuwin32/fixed/etc/Makeconf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gnuwin32/fixed/etc/Makeconf b/src/gnuwin32/fixed/etc/Makeconf
index d54c158..b8ac01a 100644
--- a/src/gnuwin32/fixed/etc/Makeconf
+++ b/src/gnuwin32/fixed/etc/Makeconf
@@ -11,8 +11,8 @@ else
   DEBUGFLAG=
 endif
 
-DLLFLAGS += -static-libgcc
-LINKFLAGS += -static-libgcc
+DLLFLAGS+= 
+LINKFLAGS+= 
 
 ## The rtools40 installer sets RTOOLS40_HOME, default to standard install path
 RTOOLS40_HOME ?= c:/rtools40
-- 
2.34.1

