From 6bd5ce080291576337567fdcc6c2bc1407961ee0 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 2 Jan 2018 15:53:10 +0000
Subject: [PATCH 09/17] macOS: include <cairo.h> not <cairo/xlib.h>

---

--- configure	2019-06-28 00:16:32.000000000 +0200
+++ configure	2019-06-28 00:16:32.000000000 +0200
@@ -40721,8 +40721,8 @@
        modlist="${modlist} cairo-svg"
        r_cairo_svg=yes
     fi
-      if "${PKG_CONFIG}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKG_CONFIG}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -40744,7 +40744,7 @@
 
 #include <pango/pango.h>
 #include <pango/pangocairo.h>
-#include <cairo-xlib.h>
+#include <cairo.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
@@ -40808,8 +40808,8 @@
          modlist="${modlist} cairo-svg"
          r_cairo_svg=yes
       fi
-      if "${PKG_CONFIG}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKG_CONFIG}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -40840,7 +40840,6 @@
 /* end confdefs.h.  */
 
 #include <cairo.h>
-#include <cairo-xlib.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
--- m4/cairo.m4	2019-03-29 00:02:40.000000000 +0100
+++ m4/cairo.m4	2019-03-29 00:02:40.000000000 +0100
@@ -52,8 +52,8 @@
        modlist="${modlist} cairo-svg"
        r_cairo_svg=yes
     fi
-      if "${PKG_CONFIG}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKG_CONFIG}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -70,7 +70,7 @@
                     [AC_LINK_IFELSE([AC_LANG_SOURCE([[
 #include <pango/pango.h>
 #include <pango/pangocairo.h>
-#include <cairo-xlib.h>
+#include <cairo.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
@@ -116,8 +116,8 @@
          modlist="${modlist} cairo-svg"
          r_cairo_svg=yes
       fi
-      if "${PKG_CONFIG}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKG_CONFIG}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -143,7 +143,6 @@
 		     [r_cv_cairo_works], 
                      [AC_LINK_IFELSE([AC_LANG_SOURCE([[
 #include <cairo.h>
-#include <cairo-xlib.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
--
3.6.1
