From 94502e90951884b82e0a2a8de35192083264ee4a Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 27 Apr 2019 00:36:24 +0100
Subject: [PATCH 16/17] Use LAPACK_LDFLAGS in Rlapack_la_LIBADD

---
 src/modules/lapack/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/modules/lapack/Makefile.in b/src/modules/lapack/Makefile.in
index 4a5e1d5..a351a53 100644
--- a/src/modules/lapack/Makefile.in
+++ b/src/modules/lapack/Makefile.in
@@ -48,7 +48,7 @@ Rlapack_la_OBJECTS = $(LIBOBJECTS)
 ## needed being loaded into R.bin/libR.so, and that need not be the
 ## case for a static libfortran or if a package is compiled under a
 ## different compiler.
-Rlapack_la_LIBADD =  $(FLIBS_IN_SO) @DYLIB_UNDEFINED_ALLOWED_FALSE@ $(LIBR)
+Rlapack_la_LIBADD =  @LAPACK_LDFLAGS@ $(FLIBS_IN_SO) @DYLIB_UNDEFINED_ALLOWED_FALSE@ $(LIBR)
 
 ALL_CFLAGS = $(ALL_CFLAGS_LO)
 ALL_FFLAGS = $(ALL_FFLAGS_LO)
