diff -Naur R-3.6.0.orig/configure R-3.6.0/configure
--- R-3.6.0.orig/configure	2019-04-18 19:16:03.000000000 -0300
+++ R-3.6.0/configure	2019-05-15 17:08:45.893564488 -0300
@@ -40686,8 +40686,8 @@
        modlist="${modlist} cairo-svg"
        r_cairo_svg=yes
     fi
-      if "${PKG_CONFIG}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKG_CONFIG}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -40709,7 +40709,7 @@
 
 #include <pango/pango.h>
 #include <pango/pangocairo.h>
-#include <cairo-xlib.h>
+#include <cairo.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
@@ -40773,8 +40773,8 @@
          modlist="${modlist} cairo-svg"
          r_cairo_svg=yes
       fi
-      if "${PKG_CONFIG}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKG_CONFIG}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -40805,7 +40805,6 @@
 /* end confdefs.h.  */
 
 #include <cairo.h>
-#include <cairo-xlib.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
diff -Naur R-3.6.0.orig/m4/cairo.m4 R-3.6.0/m4/cairo.m4
--- R-3.6.0.orig/m4/cairo.m4	2019-03-28 20:02:40.000000000 -0300
+++ R-3.6.0/m4/cairo.m4	2019-05-15 17:08:33.009310855 -0300
@@ -52,8 +52,8 @@
        modlist="${modlist} cairo-svg"
        r_cairo_svg=yes
     fi
-      if "${PKG_CONFIG}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKG_CONFIG}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -70,7 +70,7 @@
                     [AC_LINK_IFELSE([AC_LANG_SOURCE([[
 #include <pango/pango.h>
 #include <pango/pangocairo.h>
-#include <cairo-xlib.h>
+#include <cairo.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
@@ -116,8 +116,8 @@
          modlist="${modlist} cairo-svg"
          r_cairo_svg=yes
       fi
-      if "${PKG_CONFIG}" --exists cairo-xlib; then
-         xmodlist="${modlist} cairo-xlib"
+      if "${PKG_CONFIG}" --exists cairo; then
+         xmodlist="${modlist} cairo"
       else
          xmodlist="${modlist}"
       fi
@@ -143,7 +143,6 @@
 		     [r_cv_cairo_works], 
                      [AC_LINK_IFELSE([AC_LANG_SOURCE([[
 #include <cairo.h>
-#include <cairo-xlib.h>
 #if CAIRO_VERSION  < 10200
 #error cairo version >= 1.2 required
 #endif
